test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -1380,28 +1380,78 @@ public async Task NoResetOnClose(bool noResetOnClose)
                     : original));
         }
 
-        #region pgpass
-
         [Test]
         [NonParallelizable]
         public async Task UsePgPassFile()
         {
-            var builder = new NpgsqlConnectionStringBuilder(ConnectionString)
-            {
-                IntegratedSecurity = false
-            };
+            using var resetPassword = SetEnvironmentVariable("PGPASSWORD", null);
+            var builder = new NpgsqlConnectionStringBuilder(ConnectionString);
 
-            var pgpassFile = Path.GetTempFileName();
-            using var _ = Defer(() => File.Delete(pgpassFile));
-            File.WriteAllText(pgpassFile, $"*:*:*:{builder.Username}:{builder.Password}");
-            using var __ = SetEnvironmentVariable("PGPASSFILE", pgpassFile);
+            var password = builder.Password;
+            var passFile = Path.GetTempFileName();
 
-            builder.Password = null;
-            using var ___ = CreateTempPool(builder.ConnectionString, out var connectionString);
-            using var ____ = await OpenConnectionAsync(connectionString);
+            builder.Password = password;
+            builder.Passfile = passFile;
+
+            using var deletePassFile = Defer(() => File.Delete(passFile));
+
+            File.WriteAllText(passFile, $"*:*:*:{builder.Username}:{password}");
+
+            using var passFileVariable = SetEnvironmentVariable("PGPASSFILE", passFile);
+            using var pool = CreateTempPool(builder.ConnectionString, out var connectionString);
+            using var conn = await OpenConnectionAsync(connectionString);
         }
 
-        #endregion
+        [Test]
+        [NonParallelizable]
+        public void PasswordSourcePrecendence()
+        {
+            if (IsMultiplexing)
```

Comment:
Can't this run with multiplexing?

Reply:
Only the test because of multiplexing which requires pooling. Otherwise, I could play with the application name to make it work for multiplexing too.

Reply:
Not so important, it's definitely OK to leave it like this.